File: SearchingAndSorting/MergeTwoSortedArrays/MergeTwoSortedArrays.js
function method2(arr, arr2){
    let res = [];
    let n=arr.length, n2=arr2.length, i=0, j=0;
    while(i<n && j<n2){
        if(arr[i]<arr2[j]){
            res.push(arr[i]);
            i++;
        } else{
            res.push(arr2[j]);
            j++;
        }
    }
    while(i<n){
        res.push(arr[i]);
        i++;
    }

    while(j<n2){
        res.push(arr2[j])
        j++;
    }
    return res;
}

let arr = [10, 12];
let arr2 = [5, 18, 20];
console.log(method2(arr, arr2));

File: SearchingAndSorting/MergeTwoSortedArrays/MergeTwoSortedArrays.go
package main

import "fmt"

func method1(arr []int, arr2 []int) []int {
	res := make([]int, 0)
	n := len(arr)
	n2 := len(arr2)
	j := 0
	i := 0
	for i < n && j < n2 {
		if arr[i] < arr2[j] {
			res = append(res, arr[i])
			i++
		} else {
			res = append(res, arr2[j])
			j++
		}
	}
	for i < n {
		res = append(res, arr[i])
		i++
	}
	for j < n2 {
		res = append(res, arr2[j])
		j++
	}
	return res
}

func main() {
	arr := []int{1, 3, 5, 7}
	arr2 := []int{0, 2, 6, 8, 9}
	fmt.Println(method1(arr, arr2))
}
